### PR TITLE
-updated types for sort, projection and query

### PR DESCRIPTION
**Pull request by jeebeez**

### CODE REVIEW

File: src/types/collection.ts [hunk 1]
```diff
@@ -24,8 +24,7 @@ import {
   InsertOneResult,
   ListIndexesCursor,
   ListIndexesOptions,
-  MapFunction,
-  MapReduceOptions,
+  MapFunction, ObjectId,
```
Comment (tianhuil) [inline src/types/collection.ts:27]: This shouldn't happen.  Can you make sure to run prettier?

File: src/types/collection.ts [hunk 2]
```diff
@@ -465,10 +465,10 @@ export declare class SafeCollection<
    * @param options - Optional settings for the command
    * @param callback - An optional callback, a Promise will be returned if none is provided
    */
-  mapReduce<TKey = any, TValue = any>(
+  mapReduce<TKey extends ObjectId, TValue = any>(
```
Comment (tianhuil) [inline src/types/collection.ts:468]: They key doesn't have to be an `ObjectId`, if you look at the examples.

https://www.mongodb.com/docs/manual/tutorial/map-reduce-examples/

File: src/types/mapReduce.assert.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import { ObjectId } from 'mongodb'
+import * as ta from 'type-assertions'
+import { TsMapReduceOptions } from './mapReduce'
```
Comment (tianhuil) [inline src/types/mapReduce.assert.ts:3]: You should test `.mapReduce`

**Review by tianhuil — CHANGES_REQUESTED** [review]
Some changes requested.

File: src/types/mapReduce.assert.ts (hunk 3 above)
Comment (jeebeez) [reply src/types/mapReduce.assert.ts:3]: Sure, just wanted to confirm if you are referring to writing Type Assertions or Jest tests.

**Pull request closed without merging**